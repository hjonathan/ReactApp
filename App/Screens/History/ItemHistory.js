import React, {Component} from "react";
import {
    Text,
    View,
    Image
} from "react-native";
import Moment from "moment";
import I18n from "react-native-i18n";
import Card from "../../Components/Card";
import Styles from "./styles";
import * as Res from "../../Assets/resources";
import {DATETIME} from "../../Libs/Const";

class ItemHistory extends Component {
    /**
     * Constructor ItemHistory
     * @param props
     */
    constructor (props) {
        super(props);
        this.state = props.data || {
            taskName: "",
            flowStatus: "Done",
            userFullName: "",
            dueDate: "",
            userPhoto: ""
        };
        this._onLoadImage = this._onLoadImage.bind(this);
    }

    /**
     * Init status flow
     */
    componentWillMount () {
        this.formatStatusFlow();
    }

    /**
     * Executes when even load was finished
     * @private
     */
    _onLoadImage = () => {
        this.setState({loaded: true});
    };

    /**
     * Formats status flow
     */
    formatStatusFlow () {
        let status = this.state.flowStatus;
        switch (status) {
            case "DONE":
                status = I18n.t("history_activity_status_done");
                break;
            case "PAUSED":
                status = I18n.t("history_activity_status_paused");
                break;
            case "IN_PROGRESS":
                status = I18n.t("history_activity_status_inprogress");
                break;
            case "CANCELLED":
                status = I18n.t("history_activity_status_cancelled");
                break;
            case "ROUTED":
                status = I18n.t("history_activity_status_ruteado");
                break;
            default:
                status = I18n.t("history_activity_status_done");
        }
        this.setState({flowStatus: status});
    }

    /**
     * Gets First letter of first name
     * @returns {string}
     * @private
     */
    _getCapitalLetter () {
        const index = 0;
        let fullName = this.state.userFullName.split(" ").shift();
        return fullName.charAt(index).toUpperCase();
    }

    /**
     * Renders avatar
     * @returns {*}
     */
    renderImageAvatar () {
        let textDisplay,
            imageDisplay,
            backgroundColor;
        if (this.state.loaded) {
            textDisplay = "none";
            imageDisplay = "flex";
            backgroundColor = {};
        } else {
            textDisplay = "flex";
            imageDisplay = "none";
            backgroundColor = {backgroundColor: Res.colors.mainColor};
        }
        return (
            <View style={[Styles.imageSection, backgroundColor]}>
                <Image
                    onLoad={this._onLoadImage}
                    source={{uri: `data:image/gif;base64,${this.state.userPhoto}`}}
                    borderRadious={5}
                    style={[Styles.circleImage, {display: imageDisplay}]}
                />
                <Text style={{
                    color: "white",
                    fontSize: 25,
                    fontWeight: "bold",
                    display: textDisplay
                }}
                >{this._getCapitalLetter()}
                </Text>
            </View>
        );
    }

    /**
     * Renders item history
     * @returns {*}
     */
    render () {
        return (
            <Card>
                <View style={{flexDirection: "row", padding: 4, alignItems: "center"}}>
                    {this.renderImageAvatar()}
                    <View style={{paddingHorizontal: 4}}>
                        <Text style={{color: "#000", fontSize: 12}}>{this.state.taskName}</Text>
                        <Text style={{color: "#000"}}>{this.state.flowStatus}</Text>
                        <Text style={{color: "#000"}}>{this.state.userFullName}</Text>
                        <Text style={{color: "#000"}}>{Moment(this.state.dueDate).format(DATETIME.LOCAL)}</Text>
                    </View>
                </View>
            </Card>
        );
    }
}

export default ItemHistory;
