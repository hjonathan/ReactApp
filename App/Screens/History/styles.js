import {StyleSheet} from "react-native";

export default StyleSheet.create({
    mainContainer: {
        flex: 1,
        flexDirection: "row",
        padding: 8,
        backgroundColor: "#2B6697"
    },
    imageSection: {
        borderRadius: 5,
        marginHorizontal: 16,
        height: 50,
        width: 50,
        justifyContent: "center",
        alignItems: "center"
    },
    circleImageContainer: {
        alignItems: "center",
        justifyContent: "center",
        width: 50,
        height: 50
    },
    circleImage: {
        width: 50,
        height: 50,
        resizeMode: "cover"
    }
});
