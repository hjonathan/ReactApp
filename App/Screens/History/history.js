import React, {Component} from "react";
import {
    View,
    FlatList
} from "react-native";
import _ from "lodash";
import ItemHistory from "./ItemHistory";
import Styles from "./styles";
import User from "../../Model/User";
import * as Res from "../../Assets/resources";
import I18n from "../../I18n";

class History extends Component {
    /**
     * Navigation option
     */
    static navigationOptions = ({navigation}) => ({
        title: I18n.t("History"),
        headerTintColor: Res.colors.white,
        headerStyle: {
            backgroundColor: Res.colors.mainColor
        }
    });
    /**
     * Constructor
     * @param props
     */
    constructor (props) {
        super(props);
        this.state = {
            appUid: props.navigation.state.params.appUid,
            data: [],
            users: []
        };
    }

    /**
     * Init history list
     */
    componentWillMount () {
        let users = User.getUserOffline();
        this.setState({users});
        this._getHistory();
    }

    /**
     * Component will receive props
     */
    componentWillReceiveProps (nextProps) {
        try {
            if (nextProps.history.data && nextProps.history.data.flow) {
                this.mergeData(nextProps.history.data.flow);
            }
        // eslint-disable-next-line no-empty
        } catch (error) {
        }
    }

    /**
     * Merges userPhoto props to data
     * @param data
     */
    mergeData (data) {
        let userFullName,
            dataMerged,
            user;
        dataMerged = data.map((history) => {
            userFullName = history.userFullName.split(" ");
            user = _.find(this.state.users, {lastName: userFullName[0], firstName: userFullName[1]});
            if (user) {
                history.userPhoto = user.userPhoto;
            }
            return history;
        });
        this.setState({data: dataMerged});
    }

    /**
     * Request case's history.
     * @private
     */
    _getHistory () {
        this.props.requestHistory({
            app_uid: this.state.appUid
        });
    }

    /**
     * Renders screen
     * @returns {*}
     */
    render () {
        return (
            <View style={Styles.mainContainer}>
                <FlatList
                    data={this.state.data}
                    renderItem={({item}) => <ItemHistory data={item} />}
                    keyExtractor={(item, index) => index.toString()}
                />
            </View>
        );
    }
}

export default History;
