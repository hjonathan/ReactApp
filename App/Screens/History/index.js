import {connect} from "react-redux";
import History from "./history";
import actions from "../../Actions";

/**
 * Return state from Store to View Component
 * @param state
 */
const mapStateToProps = state => ({
        history: (state.Cases && state.Cases.history) || null
    }),
    mapDispatchToProps = dispatch => ({
        requestHistory: params => dispatch(actions.cases.history.request(params))
    });

export default connect(mapStateToProps, mapDispatchToProps)(History);
