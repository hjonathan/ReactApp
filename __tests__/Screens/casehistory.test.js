import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import ConnectHistory from '../../App/Screens/History';
import History from '../../App/Screens/History/history';
import ItemHistory from '../../App/Screens/History/ItemHistory';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

describe('Screen >>> History', () => {
    const initialState = {
        history: {
            data: {
                flow: [
                    {
                        taskName: 'Task 1',
                        flowStatus: 'DONE',
                        userFullName: 'Juan Perez',
                        dueDate: new Date('2016-07-01T16:25:31'),
                        userPhoto: "Picture1"
                    },
                    {
                        taskName: 'Task 2',
                        flowStatus: 'PAUSED',
                        userFullName: 'Juan Perez',
                        dueDate: new Date('2016-07-01T16:25:31'),
                        userPhoto: "Picture1"
                    },
                    {
                        taskName: 'Task 2',
                        flowStatus: 'IN_PROGRESS',
                        userFullName: 'Juan Perez',
                        dueDate: new Date('2016-07-01T16:25:31'),
                        userPhoto: "Picture1"
                    },
                    {
                        taskName: 'Task 2',
                        flowStatus: 'CANCELLED',
                        userFullName: 'Juan Perez',
                        dueDate: new Date('2016-07-01T16:25:31'),
                        userPhoto: "Picture1"
                    },
                    {
                        taskName: 'Task 2',
                        flowStatus: 'ROUTED',
                        userFullName: 'Juan Perez',
                        dueDate: new Date('2016-07-01T16:25:31'),
                        userPhoto: "Picture1"
                    }
                ]
            }
        }
    };
    const navigation = {
        state: {
            params: {
                appUid: "123456789"
            }
        }
    };
    const requestHistory = () => {};
    const mockStore = configureStore();
    let store,
        container,
        view,
        itemDone,
        itemPaused,
        itemInProgress,
        itemCancelled,
        itemRouted,
        emptyItem;

    beforeEach(() => {
        store = mockStore(initialState);
        container = shallow(<ConnectHistory store={store}/>);
        view = shallow(<History navigation={navigation} requestHistory={requestHistory}/>);
        itemDone = shallow(<ItemHistory data={initialState.history.data.flow[0]}/>);
        itemPaused = shallow(<ItemHistory data={initialState.history.data.flow[1]}/>);
        itemInProgress = shallow(<ItemHistory data={initialState.history.data.flow[2]}/>);
        itemCancelled = shallow(<ItemHistory data={initialState.history.data.flow[3]}/>);
        itemRouted = shallow(<ItemHistory data={initialState.history.data.flow[4]}/>);
        emptyItem = shallow(<ItemHistory />)
    });

    /**
     * Render test
     */
    it('Render history correctly', () => {
        expect(toJson(container)).toMatchSnapshot();
    });

    it('Render history view', () => {
        expect(toJson(view)).toMatchSnapshot();
    });

    it('Check static navigationOptions', () => {
        expect(History.navigationOptions(""))
            .toEqual({
                title: "History",
                headerTintColor: "#FFF",
                headerStyle: {backgroundColor: "#1F6CBC"}
            });
    });

    it('Render item history', () => {
        const item = {};
        expect(typeof view.props('children').children.props.renderItem).toEqual('function');
        expect(toJson(view.props('children').children.props.renderItem(item))).toMatchSnapshot();
    });

    it('Render the connected(SMART) component', () => {
        expect(container.length).toEqual(1);
    });

    it('Render history view', () => {
        view.instance().componentWillReceiveProps(initialState);
        expect(view.instance().state.data.length).toEqual(5);
    });

    it('Render item view', () => {
        expect(toJson(itemDone)).toMatchSnapshot();
        expect(toJson(itemPaused)).toMatchSnapshot();
        expect(toJson(itemInProgress)).toMatchSnapshot();
        expect(toJson(itemCancelled)).toMatchSnapshot();
        expect(toJson(itemRouted)).toMatchSnapshot();
        expect(toJson(emptyItem)).toMatchSnapshot();
    });

    it('Render item view', () => {
        expect(itemDone.instance().state.taskName).toEqual('Task 1');
        expect(itemDone.instance().state.userFullName).toEqual('Juan Perez');
        expect(itemDone.instance().state.dueDate).toEqual(new Date('2016-07-01T16:25:31'));
    });

    it('Render item view', () => {
        expect(emptyItem.instance().state.taskName).toEqual('');
        expect(emptyItem.instance().state.userFullName).toEqual('');
        expect(emptyItem.instance().state.dueDate).toEqual('');
    });
});
