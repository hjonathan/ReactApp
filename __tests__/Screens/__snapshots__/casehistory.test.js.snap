// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen >>> History Render history correctly 1`] = `
<History
  history={null}
  requestHistory={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`Screen >>> History Render history view 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#2B6697",
      "flex": 1,
      "flexDirection": "row",
      "padding": 8,
    }
  }
>
  <FlatList
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</Component>
`;

exports[`Screen >>> History Render item history 1`] = `undefined`;

exports[`Screen >>> History Render item view 1`] = `
<CardView>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "width": 50,
          },
          Object {
            "backgroundColor": "#1F6CBC",
          },
        ]
      }
    >
      <Image
        borderRadious={5}
        onLoad={[Function]}
        source={
          Object {
            "uri": "data:image/gif;base64,Picture1",
          }
        }
        style={
          Array [
            Object {
              "height": 50,
              "resizeMode": "cover",
              "width": 50,
            },
            Object {
              "display": "none",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "display": "flex",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        J
      </Text>
    </Component>
    <Component
      style={
        Object {
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
          }
        }
      >
        Task 1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        history_activity_status_done
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Juan Perez
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        2016-07-01 16:25:31
      </Text>
    </Component>
  </Component>
</CardView>
`;

exports[`Screen >>> History Render item view 2`] = `
<CardView>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "width": 50,
          },
          Object {
            "backgroundColor": "#1F6CBC",
          },
        ]
      }
    >
      <Image
        borderRadious={5}
        onLoad={[Function]}
        source={
          Object {
            "uri": "data:image/gif;base64,Picture1",
          }
        }
        style={
          Array [
            Object {
              "height": 50,
              "resizeMode": "cover",
              "width": 50,
            },
            Object {
              "display": "none",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "display": "flex",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        J
      </Text>
    </Component>
    <Component
      style={
        Object {
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
          }
        }
      >
        Task 2
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        history_activity_status_paused
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Juan Perez
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        2016-07-01 16:25:31
      </Text>
    </Component>
  </Component>
</CardView>
`;

exports[`Screen >>> History Render item view 3`] = `
<CardView>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "width": 50,
          },
          Object {
            "backgroundColor": "#1F6CBC",
          },
        ]
      }
    >
      <Image
        borderRadious={5}
        onLoad={[Function]}
        source={
          Object {
            "uri": "data:image/gif;base64,Picture1",
          }
        }
        style={
          Array [
            Object {
              "height": 50,
              "resizeMode": "cover",
              "width": 50,
            },
            Object {
              "display": "none",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "display": "flex",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        J
      </Text>
    </Component>
    <Component
      style={
        Object {
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
          }
        }
      >
        Task 2
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        history_activity_status_inprogress
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Juan Perez
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        2016-07-01 16:25:31
      </Text>
    </Component>
  </Component>
</CardView>
`;

exports[`Screen >>> History Render item view 4`] = `
<CardView>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "width": 50,
          },
          Object {
            "backgroundColor": "#1F6CBC",
          },
        ]
      }
    >
      <Image
        borderRadious={5}
        onLoad={[Function]}
        source={
          Object {
            "uri": "data:image/gif;base64,Picture1",
          }
        }
        style={
          Array [
            Object {
              "height": 50,
              "resizeMode": "cover",
              "width": 50,
            },
            Object {
              "display": "none",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "display": "flex",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        J
      </Text>
    </Component>
    <Component
      style={
        Object {
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
          }
        }
      >
        Task 2
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        history_activity_status_cancelled
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Juan Perez
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        2016-07-01 16:25:31
      </Text>
    </Component>
  </Component>
</CardView>
`;

exports[`Screen >>> History Render item view 5`] = `
<CardView>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "width": 50,
          },
          Object {
            "backgroundColor": "#1F6CBC",
          },
        ]
      }
    >
      <Image
        borderRadious={5}
        onLoad={[Function]}
        source={
          Object {
            "uri": "data:image/gif;base64,Picture1",
          }
        }
        style={
          Array [
            Object {
              "height": 50,
              "resizeMode": "cover",
              "width": 50,
            },
            Object {
              "display": "none",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "display": "flex",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        J
      </Text>
    </Component>
    <Component
      style={
        Object {
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
          }
        }
      >
        Task 2
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        history_activity_status_ruteado
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Juan Perez
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        2016-07-01 16:25:31
      </Text>
    </Component>
  </Component>
</CardView>
`;

exports[`Screen >>> History Render item view 6`] = `
<CardView>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "width": 50,
          },
          Object {
            "backgroundColor": "#1F6CBC",
          },
        ]
      }
    >
      <Image
        borderRadious={5}
        onLoad={[Function]}
        source={
          Object {
            "uri": "data:image/gif;base64,",
          }
        }
        style={
          Array [
            Object {
              "height": 50,
              "resizeMode": "cover",
              "width": 50,
            },
            Object {
              "display": "none",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "display": "flex",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        history_activity_status_done
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Invalid date
      </Text>
    </Component>
  </Component>
</CardView>
`;
